packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputSingleSelect.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridLoadIcon } from '../../icons/index';
+import { GridFilterInputValueProps } from './GridFilterInputValueProps';
+import { GridColDef } from '../../../models/colDef/gridColDef';
+
+const renderSingleSelectOptions = ({ valueOptions, valueFormatter, field }: GridColDef, api) => {
+  const iterableColumnValues =
+    typeof valueOptions === 'function'
+      ? ['', ...valueOptions({ field })]
+      : ['', ...(valueOptions || [])];
+
+  return iterableColumnValues.map((option) =>
+    typeof option === 'object' ? (
+      <option key={option.value} value={option.value}>
+        {option.label}
+      </option>
+    ) : (
+      <option key={option} value={option}>
+        {valueFormatter && option !== '' ? valueFormatter({ value: option, field, api }) : option}
+      </option>
+    ),
+  );
+};
+
+export interface GridFilterInputSingleSelectProps extends GridFilterInputValueProps {
```

Comment:
Seems that we can not extend with a union of type, so we cannot do 
```ts
export interface GridFilterInputSingleSelectProps extends GridFilterInputValueProps, TextFieldProps {
  type?: 'singleSelect';
}
```

Reply:
In that case I think we should use a type

```tsx
export type GridFilterInputSingleSelectProps = GridFilterInputValueProps & TextFieldProps & { type?: 'singleSelect' }
```

It's not great, but exporting an incomplete interface seems worse to me

Reply:
I corrected it for the two new components. I can not do it for the`GridFilterInputValue` because the type `GridFilterInputValueProps` is already exported. I add it as a comment in the breaking change issue